test/CMakeLists.txt

```diff
@@ -122,6 +122,7 @@ compile_benchmark_test(skip_with_error_test)
 add_test(NAME skip_with_error_test COMMAND skip_with_error_test --benchmark_min_time=0.01s)
 
 compile_benchmark_test(donotoptimize_test)
+target_compile_options (donotoptimize_test PRIVATE "-Wdeprecated-declarations")
```

Comment:
We really want `-Werror` too.
But, this all needs to first check with `check_cxx_compiler_flag()`.

Reply:
Done. I did both at once with `-Werror=deprecated-declarations`. If it's preferable to do `-Werror` and `-Wdeprecated-declarations` separately, I can also do so.